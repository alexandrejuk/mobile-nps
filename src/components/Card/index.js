import React from 'react'
import PropTypes from 'prop-types'
import Title from '../Title'

import styles from './style.module.css'

const Card = ({
  action,
  image,
  title,
}) => (
  <div
    onKeyPress={action}
    onClick={action}
    className={styles.card}
    tabIndex="0"
    role="button"
  >
    <div className={styles.imageContainer}>
      <img src={image} alt="item card" />
    </div>
    <div className={styles.title}>
      <Title text={title} />
    </div>
   </div>
)

Card.propTypes = {
  action: PropTypes.func.isRequired,
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}

export default Card
