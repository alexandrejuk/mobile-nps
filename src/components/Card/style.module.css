@import '../../assets/css/colors.css';

.card {
  background: var(--color-white);
  border-radius: 3px;
  box-shadow: 0px 0px 16px rgba(0, 0, 0, 0.15);
  height: 228px;
  padding: 20px;
  text-align: center;
  width: 166px;
}

.imageContainer {
  height: 164px;
  overflow: hidden;
  padding-bottom: 20px;
}

.imageContainer img {
  height: 100%;
}

.title {
  border-top: 1px solid var(--color-concrete);
  color: var(--color-ship-gray);
  font-family: 'Rubik', Arial, Helvetica, sans-serif;
  font-size: 16px;
  line-height: 14px;
  padding: 10px 0;
}
