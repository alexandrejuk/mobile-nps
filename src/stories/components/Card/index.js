import React from 'react'
import { action } from '@storybook/addon-actions'
import { Card } from '../../../components'

export default {
  title: 'Components/Card',
}

const props = {
  action: action('clicked'),
  image: 'https://www.nestleprofessional.com.br/sites/g/files/gfb331/f/styles/product/public/media/maquina-alegria-8x60-2.png?itok=3w8wjKl9',
  title: 'A8/60',
}

export const Default = () => (
  <Card {...props} />
)
