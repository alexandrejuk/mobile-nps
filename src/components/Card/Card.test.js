import React from 'react'
import { render } from '@testing-library/react'

import Card from './'

const props = {
  action: () => {},
  image: 'https://www.nestleprofessional.com.br/sites/g/files/gfb331/f/styles/product/public/media/maquina-alegria-8x60-2.png?itok=3w8wjKl9',
  title: 'A8/60',
}


describe('Card component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Card {...props} />
    )
    expect(container).toMatchSnapshot()
  })
})