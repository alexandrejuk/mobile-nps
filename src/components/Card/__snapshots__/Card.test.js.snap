// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render correctly 1`] = `
<div>
  <div
    class="card"
    role="button"
    tabindex="0"
  >
    <div
      class="imageContainer"
    >
      <img
        alt="item card"
        src="https://www.nestleprofessional.com.br/sites/g/files/gfb331/f/styles/product/public/media/maquina-alegria-8x60-2.png?itok=3w8wjKl9"
      />
    </div>
    <div
      class="title"
    >
      <h1
        class="primary medium center title normal"
      >
        A8/60
      </h1>
    </div>
  </div>
</div>
`;
